src/backend/distributed/commands/create_distributed_table.c

```diff
@@ -834,6 +829,23 @@ EnsureRelationCanBeDistributed(Oid relationId, Var *distributionColumn,
 }
 
 
+/*
+ * ErrorIfTableIsACatalogTable is a helper function to error out for citus
+ * table creation from a catalog table.
+ */
+void
+ErrorIfTableIsACatalogTable(Relation relation)
+{
+	if (relation->rd_rel->relnamespace != PG_CATALOG_NAMESPACE)
```

Comment:
Minor, feel free to ignore: Not sure that there's a lot of value in creating a function for an if statement with an error message unless we explicitly want the error message to be the same everywhere. We could give a slightly more contextual error message if the if statement is in-lined.

Reply:
I'm actually planning to move such common checks into a common function as @SaitTalhaNisanci mentioned in https://github.com/citusdata/citus/pull/4158#discussion_r488538709. So I first moved that specific check into a new function in this pr